src/main/java/br/com/metricminer2/scm/GitRemoteRepository.java

```diff
@@ -0,0 +1,182 @@
+	}
+	
+	public GitRemoteRepository(String url, String rootTempGitPath) throws InvalidRemoteException, TransportException, GitAPIException {
+		this.remoteRepositoryUrl = url;
+		this.tempGitPath = gitRemoteRepositoryTempDir(url, rootTempGitPath);
+		this.initTempGitRepository(false);
+		this.tempGitRepository = new GitRepository(tempGitPath);
+	}
+	
+	public GitRemoteRepository(String url, String rootTempGitPath, boolean bare) throws InvalidRemoteException, TransportException, GitAPIException {
+		this.remoteRepositoryUrl = url;
+		this.tempGitPath = gitRemoteRepositoryTempDir(url, rootTempGitPath);
+		this.initTempGitRepository(bare);
+		this.tempGitRepository = new GitRepository(tempGitPath);
+	}
+	
+	protected void initTempGitRepository(boolean bare) throws InvalidRemoteException, TransportException, GitAPIException {
+		File directory = new File(this.tempGitPath);
+		if(!directory.exists()) {
+			log.info("Cloning Remote Repository " + this.remoteRepositoryUrl + " into " + this.tempGitPath);
+			Git.cloneRepository()
+					.setURI(this.remoteRepositoryUrl)
+					.setBare(bare)
+					.setDirectory(directory)
+					.setCloneAllBranches(true)
+					.setNoCheckout(false)
+					.call();
+		}
+		directory.deleteOnExit();
+	}
+
+	protected static String gitSystemTempDir() {
+		return FileUtils.getTempDirectory().getAbsolutePath();
+	}
+	
+	protected static String gitRemoteRepositoryTempDir(String remoteRepositoryUrl, String rootTempDir) {
+		int lastIndexOfDotGit = remoteRepositoryUrl.lastIndexOf(".git");
+		if(lastIndexOfDotGit < 0 )
+			lastIndexOfDotGit = remoteRepositoryUrl.length();
+		String directoryName = remoteRepositoryUrl.substring(remoteRepositoryUrl.lastIndexOf("/")+1, lastIndexOfDotGit);
+
+		if(!rootTempDir.endsWith(File.separator))
+			rootTempDir += File.separator;
+		
+		return rootTempDir + directoryName;
+	}
+	
+	public static SCMRepository singleProject(String url) {
+		return singleProject(url, gitSystemTempDir());
+	}
+	
+	public static SCMRepository singleProject(String url, String rootTempGitPath) {
+		return singleProject(url, rootTempGitPath, false);
+	}
+	
+	public static SCMRepository singleProject(String url, String rootTempGitPath, boolean bare) {
+		try {
+			return new GitRemoteRepository(url, rootTempGitPath, bare).info();
+		} catch (GitAPIException e) {
+			throw new RuntimeException(e.getMessage(), e);
+		}
+	}
+	
+	public static SCMRepository[] allProjectsIn(List<String> urls, String rootTempGitPath) {
+		List<SCMRepository> repos = new ArrayList<SCMRepository>();
+		for (String url : urls) {
+			repos.add(singleProject(url, rootTempGitPath));
+		}
+
+		return repos.toArray(new SCMRepository[repos.size()]);
+	}
+
+	public static SCMRepository[] allProjectsIn(List<String> urls) {
+		List<SCMRepository> repos = new ArrayList<SCMRepository>();
+		for (String url : urls) {
+			repos.add(singleProject(url, gitSystemTempDir()));
+		}
+
+		return repos.toArray(new SCMRepository[repos.size()]);
+	}
+	
+	/**

```

Comment:
isn't this what the `deleteOnExit()` is doing in the `initTempGitRepository()` method?